account_financial_risk/models/res_partner.py

```diff
@@ -0,0 +1,554 @@
+                continue  # pragma: no cover
+            if self.property_account_receivable_id.id == account.id:
+                vals["risk_invoice_unpaid"] += self._get_amount_in_risk_currency(
+                    currency, amount_residual_currency, amount_residual, account
+                )
+            else:
+                vals["risk_account_amount_unpaid"] += self._get_amount_in_risk_currency(
+                    currency, amount_residual_currency, amount_residual, account
+                )
+        return vals
+
+    def _get_amount_in_risk_currency(
+        self, currency, amount_residual_currency, amount_residual, account
+    ):
+        acc_currency_id = currency.id
+        risk_currency_id = self.risk_currency_id.id
+        if currency.id == risk_currency_id:
+            return amount_residual_currency
+        elif acc_currency_id == risk_currency_id:
+            return amount_residual
+        return currency._convert(
+            amount_residual,
+            self.risk_currency_id,
+            self.env.company,
+            fields.Date.context_today(self),
+            round=False,
+        )
+
+    @api.depends(lambda x: x._get_depends_compute_risk_exception())
+    def _compute_risk_exception(self):
+        risk_field_list = self._risk_field_list()
+        for partner in self:
+            amount = 0.0
+            amount_exceeded = 0.0
+            risk_exception = False
+            for risk_field in risk_field_list:
+                field_value = getattr(partner, risk_field[0], 0.0)
+                max_value = getattr(partner, risk_field[1], 0.0)
+                include = getattr(partner, risk_field[2], False)
+                if max_value and field_value > max_value:
+                    risk_exception = True
+                    amount_exceeded += field_value - max_value
+                if include:
+                    amount += field_value
+            if partner.credit_limit and amount > partner.credit_limit:
+                risk_exception = True
+                amount_exceeded = amount - partner.credit_limit
+            partner.risk_total = amount
+            partner.risk_amount_exceeded = amount_exceeded
+            partner.risk_exception = risk_exception
+
+    @api.model
+    def _search_risk_exception(self, operator, value):
+        commercial_partners = self.search(
+            [
+                ("customer_rank", ">", 0),
+                "|",
+                ("is_company", "=", True),
+                ("parent_id", "=", False),
+            ],
+            order="id",
+        )
+        risk_partner_ids = commercial_partners.filtered("risk_exception").ids
+        if (operator == "=" and value) or (operator == "!=" and not value):
+            return [("id", "in", risk_partner_ids)]
+        else:
+            return [("id", "not in", risk_partner_ids)]
+
+    @api.model
+    def _max_risk_date_due(self):
+        return fields.Date.to_string(
+            fields.Date.today()
```

Comment:
Please use `context_today` to avoid issues with UTC. In LATAM, where UTC is often negative, this is more common.
```suggestion
            fields.Date.context_today(self)
```

Reply:
Done